mailpoet/lib/Automation/Engine/Registry.php

```diff
@@ -48,6 +52,34 @@ public function __construct(
     $this->steps[$rootStep->getKey()] = $rootStep;
   }
 
+  public function addTemplate(AutomationTemplate $template): void {
+    $this->templates[$template->getSlug()] = $template;
+  }
+
+  public function getTemplate(string $slug): ?AutomationTemplate {
+    return $this->getTemplates()[$slug] ?? null;
+  }
+
+  /** @return array<string, AutomationTemplate> */
+  public function getTemplates(string $category = null): array {
+    $templates = (array)$this->wordPress->applyFilters(Hooks::AUTOMATION_TEMPLATES, $this->templates);
```

Comment:
I would remove the filter and register the new templates in premium via `addTemplate()` and maybe add another method `removeTemplate(string $slug)`.

I think this would be a more streamlined approach.

What do you think?

Reply:
@websupporter Great idea! Repushed.